### PR TITLE
#17 - Action migration to composite

### LINKED ISSUE #17: Action migration to composite
Problem:
- Action is developed as node.js.
- The CPS GH actions will be developed as composite ones using Python and a shared library.

Solution:
- Migrate action to composite type.
- Adjust logic around digesting action inputs.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- Updated action.yml file.
- Renamed and updated main python file.

Closes #17

### CODE REVIEW

**Comment by miroslavpojer** [comment]
Release notes:
- Action migrated to composite type.
- Renamed main control script.
- Prepared TODO with several proposals to follow.

**Review by MobiTikula — CHANGES_REQUESTED** [review]
Besides the comments I suggest to update .gitignore file.

File: README.md [hunk 1]
```diff
@@ -233,11 +222,17 @@ For running the GitHub action incorporate these commands into the shell script a
...
-            --projects-title-filter "$PROJECTS_TITLE_FILTER" \
-            --repositories "$REPOSITORIES" \
-            --output-directory "$OUTPUT_DIRECTORY"
+python3 living_documenation_generator.py
+
+cd .. || exit 1
+```
+
+### Running single script locally
+For running just one script at the time, add the following commands to the shell script:
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply README.md:231]: Removed. Maybe it is a result of the merge.

File: src/living_documentation_generator.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    return env_vars
+
+
+def run_script(script_name, env):
+    """ Helper function to run a Python script with environment variables using subprocess """
+    try:
+        # Running the python script with given environment variables
+        result = subprocess.run(['python3', script_name], env=env, text=True, capture_output=True, check=True)
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply src/living_documentation_generator.py:21]: The usage of Python 3 should be more generic here. In the next PRs, this approach will be removed.

File: src/living_documentation_generator.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import os
+import subprocess
+import sys
+
+
+def extend_args():
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply src/living_documentation_generator.py:6]: In the current solution, we have multiple scripts that share config across system env variables. Here, we define paths in one place in the controller.

**Comment by miroslavpojer** [comment]
> Besides the comments I suggest to update .gitignore file.

What do you suppose? I do not see any change request.

**Comment by MobiTikula** [comment]
I suggest to remove `node_modules` and  `/package-lock.json`, because they keep ignored files in old versions of project, despite we have a composite action now.

**Comment by MobiTikula** [comment]
I suggest to remove `node_modules` and  `/package-lock.json`, because they keep ignored files in old versions of project, despite we have a composite action now.

Also I still see [index.js](https://github.com/AbsaOSS/living-doc-generator/blob/feature/17-action-migration-to-composite/src/index.js) in the project, is it intentionally? Composite actions should not use javascript at all via: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action and also example Composite actions.

**Comment by miroslavpojer** [comment]
> I suggest to remove `node_modules` and `/package-lock.json`, because they keep ignored files in old versions of project, despite we have a composite action now.
> 
> Also I still see [index.js](https://github.com/AbsaOSS/living-doc-generator/blob/feature/17-action-migration-to-composite/src/index.js) in the project, is it intentionally? Composite actions should not use javascript at all via: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action and also example Composite actions.

Great findings. All of them should be fixed.

**Review by MobiTikula — APPROVED** [review]

**Pull request merged**